### PR TITLE
Create Custom 404 Page

### LINKED ISSUE #2154: Create a better experience for 404 page on the Design System site.
If a user lands on a 404 page, they only have the global header and footer. The user has no means of getting back to the home page (or anywhere else) because navigation is not present.
Screen Shot 2021-12-16 at 11 26 18 AM

Create 404 page with:
- [ ] Ability to get back to the home page
- [ ] Site's navigation
- [ ] Bonus Points - 404 content with some sort of flair/personality (IT monster?, shuffling component cards, shuffling cards ending in a disaster?, other ideas)

### PR DESCRIPTION
**Pull request by taysea**
[Deploy Preview](https://deploy-preview-2232--keen-mayer-a86c8b.netlify.app/fdjkls)

#### What does this PR do?

This PR adds a custom 404 page so users can navigate back to the site. I've added some text, but thing we could likely iterate if we wanted to add a little humor/personality to the wording -- suggestions welcome.

#### Where should the reviewer start?

Type any false route. For example, 'jfdklsj'.

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [x] Small, medium, and large screen sizes
- [x] Cross-browsers (FireFox, Chrome, and Safari)
- [x] Light & dark modes
- [x] All hyperlinks route properly

**Accessibility Checks**
- [x] Keyboard interactions
- [x] Screen reader experience
- [x] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [x] Console is free of warnings and errors
- [x] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### What are the relevant issues?

Closes #2154 

#### Is this change backwards compatible or is it a breaking change?
Yes, backwards compatible.

### CODE REVIEW

File: aries-site/src/pages/404.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    </Box>
+  );
+};
+
+const Message = () => (
+  <Box align="start" gap="small">
+    <Heading margin="none">Oops, there’s nothing here.</Heading>
+    <Paragraph margin={{ bottom: 'medium', top: 'none' }} size="large">
+      The page you’re looking for could not be found. Error Type: 404.
+    </Paragraph>
```
Comment (halocline) [inline aries-site/src/pages/404.js:36]: Just a suggestion... soft opinion.

```suggestion
    <Paragraph margin={{ bottom: 'medium', top: 'none' }} size="large">
      The page you’re looking for could not be found.
    </Paragraph>
    <Paragraph margin={{ bottom: 'medium', top: 'none' }} size="large">
      Error Type: 404.
    </Paragraph>
```

**Review by halocline — APPROVED** [review]
Thank you for adding some personality! I appreciate the extra effort.

**Review by MikeKingdom — APPROVED** [review]
Looks great. I think we need a gremlin overseeing the text :)

**Review by britt6612 — APPROVED** [review]
Much better!

File: aries-site/src/pages/404.js (hunk 1 above)
Comment (taysea) [reply aries-site/src/pages/404.js]: Cool, I've broken it into 2 paragraphs.

**Comment by bahriaditi** [comment]
looks good to me - thanks @taysea!

**Pull request merged**